### PR TITLE
Anushka/random controller

### PR DESCRIPTION
**Pull request by nusshkii**
Defined the random button function

### CODE REVIEW

File: src/components/header/random-controller/index.tsx [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+    const theme = useThemeStore(set => set.theme)
+    const soundsRef = useRef(sounds)
+    const intervalIdRef = useRef(null)
+  
+    useEffect(() => {
+      soundsRef.current = sounds
+    }, [sounds])
+  
+    function randomizeVolumes() {
+      soundsRef.current
```
Comment (Inverseit) [inline src/components/header/random-controller/index.tsx:23]: Will eventually have to change this.

File: src/components/header/random-controller/index.tsx [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+        intervalIdRef.current = setInterval(randomizeVolumes, updateInterval)
+      } else {
+        clearInterval(intervalIdRef.current)
+        intervalIdRef.current = null
+      }
+  
+      return () => {
+        clearInterval(intervalIdRef.current)
+      }
+    }, [randomMode, updateInterval])
```
Comment (Inverseit) [inline src/components/header/random-controller/index.tsx:46]: This is cool

File: src/components/header/random-controller/index.tsx [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+  
+    return (
+      <button
+        title="Enable/Disable Random Mode"
+        className={controllerButton({ theme })}
+        onClick={() => setRandomMode(!randomMode)}
+        data-umami-event="Enable/Disable Random Mode"
+      >
+        {randomMode ? <FiZapOff size={25} /> : <FiZap size={25} />}
```
Comment (Inverseit) [inline src/components/header/random-controller/index.tsx:55]: good icon

**Review by Inverseit — APPROVED** [review]
LGTM

File: src/components/header/random-controller/index.tsx (hunk 3 above)
Comment (nusshkii) [reply src/components/header/random-controller/index.tsx:55]: thank you!

**Pull request merged**